Rubberduck.Parsing/ComReflection/ComTypedElement.cs

```diff
@@ -116,49 +116,49 @@ private void GetValue(IntPtr lpvarValue)
+                GetFieldType(Marshal.PtrToStructure<TYPEDESC>(desc.lpValue), info);
+            }
+            else if (Convert.ToBoolean(desc.vt & (int)VarEnum.VT_USERDEFINED))
+            {
+                int href;
+                unchecked
+                {
+                    href = (int)(desc.lpValue.ToInt64() & 0xFFFFFFFF);
+                }
+                try
+                {
+                    info.GetRefTypeInfo(href, out ITypeInfo refTypeInfo);
+                    refTypeInfo.GetTypeAttr(out IntPtr attribPtr);
+                    using (DisposalActionContainer.Create(attribPtr, refTypeInfo.ReleaseTypeAttr))
                     {
-                        info.GetRefTypeInfo(href, out ITypeInfo refTypeInfo);
-                        refTypeInfo.GetTypeAttr(out IntPtr attribPtr);
-                        using (DisposalActionContainer.Create(attribPtr, refTypeInfo.ReleaseTypeAttr))
+                        var attribs = Marshal.PtrToStructure<TYPEATTR>(attribPtr);
+                        if (attribs.typekind == TYPEKIND.TKIND_ENUM)
                         {
-                            var attribs = Marshal.PtrToStructure<TYPEATTR>(attribPtr);
-                            if (attribs.typekind == TYPEKIND.TKIND_ENUM)
-                            {
-                                EnumGuid = attribs.guid;
-                            }
-                            else if (attribs.typekind == TYPEKIND.TKIND_ALIAS)
-                            {
-                                AliasGuid = attribs.guid;
-                            }
-                            IsReferenceType = ReferenceTypeKinds.Contains(attribs.typekind);
-                            _type = new ComDocumentation(refTypeInfo, -1).Name;
+                            EnumGuid = attribs.guid;
                         }
+                        else if (attribs.typekind == TYPEKIND.TKIND_ALIAS)
+                        {
+                            AliasGuid = attribs.guid;
+                        }
+                        IsReferenceType = ReferenceTypeKinds.Contains(attribs.typekind);
+                        _type = new ComDocumentation(refTypeInfo, -1).Name;
                     }
-                    catch (COMException) { }
-                    break;
-                case VarEnum.VT_SAFEARRAY:
-                case VarEnum.VT_CARRAY:
-                case VarEnum.VT_ARRAY:
-                    GetFieldType(Marshal.PtrToStructure<TYPEDESC>(desc.lpValue), info);
-                    IsArray = true;
-                    break;
-                default:
-                    _type = ComVariant.TypeNames.TryGetValue((VarEnum)desc.vt, out var typeName) ? typeName : "Object";
-                    break;
+                }
+                catch (COMException) { }
+            }
+            else if (Convert.ToBoolean(desc.vt & (int)VarEnum.VT_SAFEARRAY) ||
```

Comment:
and here, but note the handling for VT_CARRAY and VT_ARRAY on the next line is correct, as they are proper flags.

Reply:
That's basically a shortcut on my part - the reason I wasn't filtering for the basic types was because anything that is a pointer should be recursing until it is a basic type.